app/controllers/mandatarissen/search.js

```diff
@@ -42,7 +42,7 @@ export default class MandatarissenSearchController extends Controller {
 
   @action
   selectPeriod(period) {
-    this.bestuursperiode = period.id;
+    this.bestuursperiode = period.period.id;
   }
```

Comment:
Little strange that it is periode.period? Maybe as you did below also change it to periodMap? periodMap.period.id